### ISSUE TITLE
[JIT] integer constant handling seems inconsistent

### ISSUE THREAD

**Issue by gchanan** [pytorch/pytorch#6714]
After removing some zero-dim handling in ATen (https://github.com/pytorch/pytorch/pull/6683), I get a test failure in `python test/test_jit.py TestScript.test_script_module_for`.   The error given is:

```
======================================================================
ERROR: test_script_module_for (__main__.TestScript)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_jit.py", line 2158, in test_script_module_for
    self.assertEqual(m(), 10)
  File "/data/users/gchanan/pytorch9j/torch/nn/modules/module.py", line 460, in __call__
    result = self.forward(*input, **kwargs)
  File "/data/users/gchanan/pytorch9j/torch/jit/__init__.py", line 761, in forward
    return self.__getattr__('forward')(*args, **kwargs)
RuntimeError:
Expected object of type CPULongType but found type CPUIntType for argument #3 'other' (checked_cast_tensor at ../src/ATen/Utils.h:30)
frame #0: at::CPULongTensor* at::checked_cast_tensor<at::CPULongTensor, at::TensorImpl>(at::TensorImpl*, char const*, int, bool) + 0x110 (0x7f8613781106 in /data/users/gchanan/pytorch9j/torch/lib/libATen.so)
frame #1: at::CPULongType::s_add(at::Tensor const&, at::Tensor const&, at::Scalar) const + 0x152 (0x7f861382d240 in /data/users/gchanan/pytorch9j/torch/lib/libATen.so)
frame #2: torch::autograd::VariableType::s_add(at::Tensor const&, at::Tensor const&, at::Scalar) const + 0x2e9 (0x7f86159403cb in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #3: at::Type::add(at::Tensor const&, at::Tensor const&, at::Scalar) const + 0xda (0x7f861387099c in /data/users/gchanan/pytorch9j/torch/lib/libATen.so)
frame #4: <unknown function> + 0xed673c (0x7f861560a73c in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #5: <unknown function> + 0xefa28c (0x7f861562e28c in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #6: <unknown function> + 0xff8e63 (0x7f861572ce63 in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #7: std::function<int (std::vector<at::Tensor, std::allocator<at::Tensor> >&)>::operator()(std::vector<at::Tensor, std::allocator<at::Tensor> >&) const + 0x49 (0x7f86154a7c1f in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #8: torch::jit::InterpreterStateImpl::runOneStage(std::vector<at::Tensor, std::allocator<at::Tensor> >&) + 0xb2 (0x7f86154a3e2c in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #9: torch::jit::InterpreterState::runOneStage(std::vector<at::Tensor, std::allocator<at::Tensor> >&) + 0x2b (0x7f861549b1d1 in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #10: torch::jit::GraphExecutorImpl::runFallback(torch::jit::variable_tensor_list) + 0x69 (0x7f86154ef411 in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #11: torch::jit::GraphExecutorImpl::run(torch::jit::variable_tensor_list) + 0x219 (0x7f86154eef05 in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #12: torch::jit::GraphExecutor::run(torch::jit::variable_tensor_list&&) + 0x55 (0x7f86154eaa51 in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #13: torch::jit::script::Method::run(torch::jit::variable_tensor_list&&) + 0xd0 (0x7f8615860dc8 in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #14: <unknown function> + 0x1127035 (0x7f861585b035 in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #15: <unknown function> + 0x112c873 (0x7f8615860873 in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #16: <unknown function> + 0x112c0c3 (0x7f86158600c3 in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #17: <unknown function> + 0x112b79a (0x7f861585f79a in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #18: <unknown function> + 0x112b82b (0x7f861585f82b in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #19: <unknown function> + 0xc7fcf4 (0x7f86153b3cf4 in /data/users/gchanan/pytorch9j/torch/_C.cpython-35m-x86_64-linux-gnu.so)
frame #20: PyCFunction_Call + 0x131 (0x7f8629dbedd1 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #21: PyObject_Call + 0x56 (0x7f8629d694c6 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #22: <unknown function> + 0x83bdc (0x7f8629d85bdc in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #23: PyObject_Call + 0x56 (0x7f8629d694c6 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #24: <unknown function> + 0xdb25f (0x7f8629ddd25f in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #25: PyObject_Call + 0x56 (0x7f8629d694c6 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #26: PyEval_EvalFrameEx + 0x6276 (0x7f8629e44286 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #27: <unknown function> + 0x145fc9 (0x7f8629e47fc9 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #28: PyEval_EvalCodeEx + 0x48 (0x7f8629e48158 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #29: <unknown function> + 0x9ac91 (0x7f8629d9cc91 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #30: PyObject_Call + 0x56 (0x7f8629d694c6 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #31: PyEval_EvalFrameEx + 0x6276 (0x7f8629e44286 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #32: <unknown function> + 0x145fc9 (0x7f8629e47fc9 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #33: PyEval_EvalCodeEx + 0x48 (0x7f8629e48158 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #34: <unknown function> + 0x9ab72 (0x7f8629d9cb72 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #35: PyObject_Call + 0x56 (0x7f8629d694c6 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #36: <unknown function> + 0x83bdc (0x7f8629d85bdc in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #37: PyObject_Call + 0x56 (0x7f8629d694c6 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #38: <unknown function> + 0xdb25f (0x7f8629ddd25f in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #39: PyObject_Call + 0x56 (0x7f8629d694c6 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #40: PyEval_EvalFrameEx + 0x778c (0x7f8629e4579c in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #41: <unknown function> + 0x145fc9 (0x7f8629e47fc9 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #42: PyEval_EvalFrameEx + 0x812d (0x7f8629e4613d in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #43: <unknown function> + 0x145fc9 (0x7f8629e47fc9 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #44: PyEval_EvalCodeEx + 0x48 (0x7f8629e48158 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #45: <unknown function> + 0x9ac91 (0x7f8629d9cc91 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #46: PyObject_Call + 0x56 (0x7f8629d694c6 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #47: PyEval_EvalFrameEx + 0x6276 (0x7f8629e44286 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #48: <unknown function> + 0x145fc9 (0x7f8629e47fc9 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #49: PyEval_EvalCodeEx + 0x48 (0x7f8629e48158 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #50: <unknown function> + 0x9ab72 (0x7f8629d9cb72 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #51: PyObject_Call + 0x56 (0x7f8629d694c6 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #52: <unknown function> + 0x83bdc (0x7f8629d85bdc in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #53: PyObject_Call + 0x56 (0x7f8629d694c6 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #54: <unknown function> + 0xdb25f (0x7f8629ddd25f in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #55: PyObject_Call + 0x56 (0x7f8629d694c6 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #56: PyEval_EvalFrameEx + 0x778c (0x7f8629e4579c in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #57: <unknown function> + 0x145fc9 (0x7f8629e47fc9 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #58: PyEval_EvalCodeEx + 0x48 (0x7f8629e48158 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #59: <unknown function> + 0x9ac91 (0x7f8629d9cc91 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #60: PyObject_Call + 0x56 (0x7f8629d694c6 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #61: PyEval_EvalFrameEx + 0x6276 (0x7f8629e44286 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #62: <unknown function> + 0x145fc9 (0x7f8629e47fc9 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
frame #63: PyEval_EvalCodeEx + 0x48 (0x7f8629e48158 in /data/users/gchanan/anaconda3/envs/py35b/bin/../lib/libpython3.5m.so.1.0)
:
operation failed in interpreter:
@torch.jit.script_method
def forward(self):
    sum = 0
    for i in self.b:
        sum += i
        ~~~~~~~~ <--- HERE
    return sum
```

This seems fishy because everything is a python (long) constant here and the test passes if I turn all the constants in floats.

CC @apaszke @zdevito.

**Comment by zdevito** [comment]
Here is the fix: https://github.com/pytorch/pytorch/pull/6728

**Comment by zou3519** [comment]
This looks fixed by #6728 so I am closing this.
